### PR TITLE
Indexing sbt plugins

### PR DESCRIPTION
**Pull request by julienrf**
This is a work in progress for #191.

- [x] fetch sbt plugins from bintray ;
- [x] save relevant data in the scaladex-index repository ;
- [x] also save ivy.xml descriptors in the scaladex-index repository ;
- [x] include sbt plugins in the elastic search indexing process ;
- [x] download github information for sbt-plugins based on claims ;
- [x] show `addSbtPlugin(…)` instructions instead of `libraryDependencies += …` in the case of sbt plugins.

Design decisions:
- `MavenModel` has been generalized to `ReleaseModel` ;
- `LocalRepository` has been specialized to `LocalPomRepository`.

### CODE REVIEW

File: data/src/main/scala/ch.epfl.scala.index.data/bintray/BintrayDownloadSbtPlugins.scala [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+  private def parseRelease(release: SbtPluginReleaseWithModuleDescriptor): SbtPluginReleaseModel = {
+    val descriptor = release.moduleDescriptor
+    val releaseModel =
+      ReleaseModel(
+        groupId = release.sbtRelease.organization,
+        artifactId = release.sbtRelease.artifact,
+        version = release.sbtRelease.version,
+        packaging = "jar",
+        name = None,
+        description = Some(descriptor.getDescription),
```
Comment (MasseGuillaume) [inline data/src/main/scala/ch.epfl.scala.index.data/bintray/BintrayDownloadSbtPlugins.scala]: Option() for null

File: data/src/main/scala/ch.epfl.scala.index.data/bintray/BintrayDownloadSbtPlugins.scala [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+      (oldReleases, lastDownload) <- Future.successful(ivysData.read())
+      newReleases <- fetchNewReleases(lastDownload)
+    } yield {
+      ivysData.update(oldReleases, newReleases.map(parseRelease))
+    }
+
+  private def parseRelease(release: SbtPluginReleaseWithModuleDescriptor): SbtPluginReleaseModel = {
+    val descriptor = release.moduleDescriptor
+    val releaseModel =
+      ReleaseModel(
```
Comment (MasseGuillaume) [inline data/src/main/scala/ch.epfl.scala.index.data/bintray/BintrayDownloadSbtPlugins.scala]: Check for nulls this is a java api

**Comment by julienrf** [comment]
I think this is ready to merge.

**Comment by julienrf** [comment]
We still have to update the claims.json to link sbt plugins to their github project page.

File: data/src/main/scala/ch.epfl.scala.index.data/download/PlayWsDownloader.scala [hunk 3]
```diff
@@ -47,6 +44,20 @@ trait PlayWsDownloader {
...
   /**
+    * Creates a fresh client and closes it after the future returned by `f` completes.
+    *
+    * {{{
+    *   managed { client =>
+    *     client.url("http://google.com").get()
+    *   }
+    * }}}
+    */
+  def managed[A](f: WSClient => Future[A])(implicit ec: ExecutionContext): Future[A] = {
```
Comment (MasseGuillaume) [inline data/src/main/scala/ch.epfl.scala.index.data/download/PlayWsDownloader.scala]: S2

Comment (julienrf) [inline data/src/main/scala/ch.epfl.scala.index.data/download/PlayWsDownloader.scala]: Elaborate?

Comment (MasseGuillaume) [inline data/src/main/scala/ch.epfl.scala.index.data/download/PlayWsDownloader.scala]: Try/Catch ?

Comment (MasseGuillaume) [inline data/src/main/scala/ch.epfl.scala.index.data/download/PlayWsDownloader.scala]: S2 = :heart:

**Comment by MasseGuillaume** [comment]
Testing locally and updating claims

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -30,15 +34,47 @@ If you have an elasticsearch service installed use the following sbt command whe
 
 ## Overview
 
-![](http://knsv.github.io/mermaid/live_editor/#/view/Z3JhcGggTFIKICBqZW5raW5zIC0tIGluZGV4aW5nIGpvYiAtLT4gaW5kZXguZ2l0CiAgdXNlcnMvc29uYXR5cGUgLS0gZWRpdCBQT01zIG9yIHByb2plY3RzIC0tPiBzY2FsYWRleAogIGluZGV4LmdpdAogIHVzZXJzL3NvbmF0eXBlIC0tIGNsYWltIHByb2plY3RzIC0tPiBjb250cmliLmdpdAogIHNjYWxhZGV4IC0tIHdyaXRlIGVkaXRzIC0tPiBpbmRleC5naXQKICBzY2FsYWRleCAtLSBmdWxsIHRleHQgc2VhcmNoIC0tPiBlbGFzdGljLXNlYXJjaAogIHNjYWxhZGV4IC0tIHJlYWQgY2xhaW1zIC0tPiBjb250cmliLmdpdA)
+```
```
Comment (jvican) [inline CONTRIBUTING.md]: This is so awesome.

Comment (MasseGuillaume) [inline CONTRIBUTING.md]: hehe

**Pull request merged**